### PR TITLE
ref(grouping): Change defaults and improve endpoint

### PR DESCRIPTION
**Pull request by armenzg**
Changes included:
* Prevent calling the endpoint with the wrong parameters
* Prevent mixing different modes of execution
* Prevent starting backfills without "skip_processed_projects"

### CODE REVIEW

File: src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py [hunk 1]
```diff
@@ -22,36 +22,53 @@ class ProjectBackfillSimilarIssuesEmbeddingsRecords(ProjectEndpoint):
...
+            else None
+        )
 
-        if request.data.get("skip_processed_projects"):
-            skip_processed_projects = True
+        # These overwrite the defaults of backfill_seer_grouping_records_for_project
+        only_delete = True if request.data.get("only_delete") else False
+        enable_ingestion = True if request.data.get("enable_ingestion") else False
+        # Notice that it reads as "process" rather than "skip", thus, it's the opposite of what you'd expect
+        skip_processed_projects = True if request.data.get("process_skipped_projects") else False
```
Comment (armenzg) [inline src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py:65]: This removes remembering to specify "skip_processed_projects" when starting a backfill.

File: src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py
Comment (armenzg) [inline src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py:1]: When do we use this endpoint?

File: src/sentry/tasks/embeddings_grouping/backfill_seer_grouping_records_for_project.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def backfill_seer_grouping_records_for_project(
     last_processed_project_index_input: int | None = None,
     only_delete: bool = False,
     enable_ingestion: bool = False,
-    skip_processed_projects: bool = False,
+    skip_processed_projects: bool = True,
```
Comment (armenzg) [inline src/sentry/tasks/embeddings_grouping/backfill_seer_grouping_records_for_project.py:56]: This prevents forgetting to specify the value.

File: src/sentry/tasks/embeddings_grouping/backfill_seer_grouping_records_for_project.py [hunk 3]
```diff
@@ -297,16 +295,16 @@ def backfill_seer_grouping_records_for_project(
...
+    enable_ingestion: bool,
+    skip_processed_projects: bool,
+    skip_project_ids: list[int] | None,
+    worker_number: int | None,
     only_delete: bool = False,
-    enable_ingestion: bool = False,
-    skip_processed_projects: bool = False,
-    skip_project_ids: list[int] | None = None,
+    last_processed_group_id: int | None = None,
     last_processed_project_id: int | None = None,
```
Comment (armenzg) [inline src/sentry/tasks/embeddings_grouping/backfill_seer_grouping_records_for_project.py:307]: Only these last two parameters will be set to a default.

File: src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py [hunk 4]
```diff
@@ -22,36 +22,53 @@ class ProjectBackfillSimilarIssuesEmbeddingsRecords(ProjectEndpoint):
...
-
-        if request.data.get("enable_ingestion"):
-            enable_ingestion = True
+        last_processed_id = (
+            int(request.data["last_processed_id"])
+            if request.data.get("last_processed_id")
+            else None
+        )
+        skip_project_ids = (
+            [int(i) for i in request.data["skip_project_ids"].split(",")]
```
Comment (armenzg) [inline src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py:56]: Is this necessary? Does Django automatically turn a string of numbers into a list of numbers?

File: src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py [hunk 5]
```diff
@@ -22,36 +22,53 @@ class ProjectBackfillSimilarIssuesEmbeddingsRecords(ProjectEndpoint):
...
 
-        if request.data.get("skip_processed_projects"):
-            skip_processed_projects = True
+        # These overwrite the defaults of backfill_seer_grouping_records_for_project
+        only_delete = True if request.data.get("only_delete") else False
+        enable_ingestion = True if request.data.get("enable_ingestion") else False
+        # Notice that it reads as "reprocess" rather than "skip", thus, it's the opposite of what you'd expect
+        skip_processed_projects = (
+            False if request.data.get("reprocess_backfilled_projects") else True
+        )
```
Comment (armenzg) [inline src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py:65]: You're right. I inverted the values to overwrite the default values.
```
    only_delete: bool = False --> True
    enable_ingestion: bool = False --> True
    skip_processed_projects: bool = True --> False
```

**Comment by armenzg** [comment]
I have addressed your feedback.
I will look into test failures later.

**Review by jangjodi — APPROVED** [review]

**Pull request merged**